File: Assignments/03a [Individual] Data parsing server - Part III/02._python/parse.py
import os
import json
import csv
import yaml
import xml.etree.ElementTree as ET

DATA_FOLDER = r"C:\Users\nicla\OneDrive\Documents\SystemIntegration\SoftwareIntegration\Assignments\03a [Individual] Data parsing server - Part III\data"

def parse_txt(file_path):
    with open(file_path, "r", encoding="utf-8") as f:
        return f.read().splitlines()

def parse_json(file_path):
    with open(file_path, "r", encoding="utf-8") as f:
        return json.load(f)

def parse_yaml(file_path):
    with open(file_path, "r", encoding="utf-8") as f:
        return yaml.safe_load(f)

def parse_csv(file_path):
    with open(file_path, "r", encoding="utf-8") as f:
        reader = csv.DictReader(f)
        return [row for row in reader]

def parse_xml(file_path):
    tree = ET.parse(file_path)
    root = tree.getroot()
    return [elem.text for elem in root.findall("fruit")]

PARSERS = {
    "txt": parse_txt,
    "json": parse_json,
    "yaml": parse_yaml,
    "csv": parse_csv,
    "xml": parse_xml
}

File: 00._Assignments/03a [Individual] Data parsing server - Part III/01._node/parse.js
import fs from "fs";
import yaml from "yaml";
import { parseStringPromise } from "xml2js";

export function parseTxt(filePath) {
    return fs.readFileSync(filePath, "utf-8").split("\n").map(line => line.trim()).filter(Boolean);
}

export function parseJson(filePath) {
    return JSON.parse(fs.readFileSync(filePath, "utf-8"));
}

export function parseYaml(filePath) {
    return yaml.parse(fs.readFileSync(filePath, "utf-8"));
}

export function parseCsv(filePath) {
    const data = fs.readFileSync(filePath, "utf-8");
    const lines = data.split("\n").map(line => line.trim()).filter(Boolean);
    const headers = lines.shift().split(",");
    return lines.map(line => {
        const values = line.split(",");
        return headers.reduce((obj, key, i) => ({ ...obj, [key]: values[i] }), {});
    });
}

export async function parseXml(filePath) {
    const data = fs.readFileSync(filePath, "utf-8");
    const result = await parseStringPromise(data);
    return result.fruits.fruit;
}
